File: Java/dependency-injection/service-locator/src/main/java/ServiceLocatorAntipattern/Inbound/BinaryParsing.java
package ServiceLocatorAntipattern.Inbound;

import ServiceLocatorAntipattern.ApplicationRoot;
import ServiceLocatorAntipattern.InMessages.NullMessage;
import ServiceLocatorAntipattern.InMessages.StartMessage;
import ServiceLocatorAntipattern.InMessages.StopMessage;
import ServiceLocatorAntipattern.Interfaces.AcmeMessage;

public class BinaryParsing implements PacketParsing {
  public AcmeMessage resultFor(byte[] frameData) {
    if (frameData == null) {
      return ApplicationRoot.context.getComponent(NullMessage.class);
    } else if (frameData[0] == 1) {
      return ApplicationRoot.context.getComponent(StartMessage.class);
    } else {
      return ApplicationRoot.context.getComponent(StopMessage.class);
    }
  }
}

File: CSharp/DependencyInjection/DependencyInjection/DependencyInjection/_1_ControlFreak/Inbound/BinaryParsing.cs
using DependencyInjectionBefore._1_ControlFreak.InMessages;
using DependencyInjectionBefore._1_ControlFreak.Interfaces;

namespace DependencyInjectionBefore._1_ControlFreak.Inbound
{
  class BinaryParsing
  {
    public AcmeMessage ResultFor(byte[] frameData)
    {
      if (frameData == null)
      {
        return new NullMessage();
      }
      else if (frameData[0] == 1)
      {
        return new StartMessage();
      }
      else
      {
        return new StopMessage();
      }
      
    }
  }
}